Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -1293,13 +1293,14 @@ template <class TreeBuilder> TreeDestructuringPattern Parser<LexerType>::parseDe
             JSTokenLocation location = m_token.m_location;
             bool escapedKeyword = match(ESCAPED_KEYWORD);
             if (escapedKeyword || matchSpecIdentifier()) {
-                failIfTrue(match(LET) && (kind == DestructuringKind::DestructureToLet || kind == DestructuringKind::DestructureToConst), "Cannot use 'let' as an identifier name for a LexicalDeclaration");
+                bool letMatched = match(LET);
                 propertyName = m_token.m_data.ident;
                 JSToken identifierToken = m_token;
                 next();
                 if (consume(COLON))
                     innerPattern = parseBindingOrAssignmentElement(context, kind, exportType, duplicateIdentifier, hasDestructuringPattern, bindingContext, depth + 1);
                 else {
+                    semanticFailIfTrue(letMatched && (kind == DestructuringKind::DestructureToLet || kind == DestructuringKind::DestructureToConst), "Cannot use the keyword 'let' as a lexical variable name");
```

Comment:
I wonder if we could separate out the binding target validation/error reporting from parseBindingOrAssignmentElement, and reuse it here, to ensure that all the same things are checked. But otherwise this seems fine to me.